projects/sandbox/train/train/data_structures.py

```diff
@@ -8,6 +8,33 @@
 from ml4gw.utils.slicing import sample_kernels
 
 
+class ChannelSwapper(torch.nn.Module):
+    """
+    Data augmentation module that randomly swaps channels
+    of a fraction of batch elements.
+
+    Args:
+        frac:
+            Fraction of batch that will have channels swapped.
+    """
+
+    def __init__(self, frac: float = 0.5):
+        super().__init__()
+        self.frac = frac
+
+    def forward(self, X):
+        num = int(X.shape[0] * self.frac)
+        indices = None
+        if num > 0:
+            num = num if not num % 2 else num + 1
+            channel = torch.randint(X.shape[1], size=(num // 2,)).repeat(2)
+            indices = torch.randperm(X.shape[0])[:num]
```

Comment:
Not a big deal, especially if this already works, but I actually don't think we need to be super diligent about randomness here due to the fact that gradient descent just averages over the batch:
- You can swap the same channel each time, it's functionally the same as swapping the other channel then swapping the batch elements
- You can just swap the first `num / 2` indices with the next `num / 2` indices since our waveform sampling and background sampling are both uniform (i.e. have no dependence on their position within a batch)

But this is obviously more general, so maybe it's worth keeping.

Reply:
This makes things much clearer so all for it

Reply:
Can I ask why not the same argument for the `Signalnverter` and `SignalReverser`?